jax/experimental/export/_export.py

```diff
@@ -307,6 +307,17 @@ def __str__(self):
     # do not want the entire serialized module to end up in locations.
     return f"Exported(fun_name={self.fun_name}, ...)"
 
+  def in_shardings_for_mesh(self,
+                            mesh: sharding.Mesh) -> Sequence[sharding.XLACompatibleSharding | None]:
+    """Create XLACompatibleSharding for the inputs.
+
+    `in_shardings` are HloShardings, which are serializable because they are
+    independent of the mesh and the device assignment.
+    This method turns them into XLACompatibleSharding for the given mesh.
+    """
+    return tuple(sharding.GSPMDSharding(mesh.devices, s) if s is not None else None
```

Comment:
It seems weird to return GSPMDSharding when you have a mesh.

Can we return NamedSharding instead? You can use `pxla._gspmd_to_named_sharding_via_mesh(gspmd_sharding, mesh)` to get a NamedSharding back

Reply:
We discussed offline and we agreed to change the code to return NamedSharding, even though in some rare situations it may be possible that we cannot construct NamedSharding from an arbitrary HloSharding. This can only happen if the HloSharding was generated with a mesh of a different shape than the mesh we want to use for running.